### PR TITLE
Ginger Runset Execution Report DB Link to ALM

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to export a report link when publishing to ALM.

- **Improvements**
  - Enhanced the layout of menu items by adjusting margins for better visual alignment in the main window.

- **Bug Fixes**
  - Refined exception handling for image download methods to improve stability.

- **Chores**
  - Updated project references to use local DLLs instead of package references for `Ginger.ExecuterService.Contracts`.

### CODE REVIEW

File: Ginger/GingerCoreCommon/Run/RunSetActions/RunSetActionPublishToQC.cs [hunk 1]
```diff
@@ -44,6 +44,25 @@ public class RunSetActionPublishToQC : RunSetActionBase, INotifyPropertyChanged
         [IsSerializedForLocalRepository]
         public bool toAttachActivitiesGroupReport { get { return mtoAttachActivitiesGroupReport; } set { if (mtoAttachActivitiesGroupReport != value) { mtoAttachActivitiesGroupReport = value; OnPropertyChanged(nameof(toAttachActivitiesGroupReport)); } } }
 
+
+        private bool mtoExportReportLink;
+        [IsSerializedForLocalRepository]
+        public bool toExportReportLink
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Run/RunSetActions/RunSetActionPublishToQC.cs:50]: change name to ToExportReportLink

File: Ginger/GingerCoreNET/ALMLib/JIRA/Bll/JiraExportManager.cs [hunk 2]
```diff
@@ -369,6 +372,22 @@ public bool ExecuteDataToJira(BusinessFlow bizFlow, PublishToALMConfig publishTo
                                 System.IO.File.Delete(zipFileName);
                             }
                         }
+
+
+                        if (publishToALMConfig.ToExportReportLink)
+                        {
+                            if (!string.IsNullOrEmpty(publishToALMConfig.HtmlReportUrl))
+                            {
+                                string reportLink = publishToALMConfig.HtmlReportUrl + "#/BusinessFlow/" + $"{ publishToALMConfig.ExecutionId}/" + bizFlow.InstanceGuid;
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/ALMLib/JIRA/Bll/JiraExportManager.cs:381]: Use the common function to create report URL

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreCommon/Run/RunSetActions/RunSetActionPublishToQC.cs (hunk 1 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/Run/RunSetActions/RunSetActionPublishToQC.cs]: handled

File: Ginger/GingerCoreNET/ALMLib/JIRA/Bll/JiraExportManager.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/ALMLib/JIRA/Bll/JiraExportManager.cs]: handled

**Pull request merged**